api/src/test/java/uk/gov/dhsc/htbhf/claimant/entitlement/PaymentCycleEntitlementCalculatorTest.java

```diff
@@ -108,6 +108,20 @@ void shouldCallEntitlementCalculatorWithExpectedDueDateWhenNoPreviousPregnancyVo
         verifyEntitlementCalculatorCalled(expectedDueDate, dateOfBirthsOfChildren);
     }
 
+    @Test
+    void shouldCallEntitlementCalculatorWithExpectedDueDateWhenPreviousEntitlementIsNullDueToIneligibilityInPreviousCycle() {
+        VoucherEntitlement voucherEntitlement = aValidVoucherEntitlement();
+        given(entitlementCalculator.calculateVoucherEntitlement(any(), any(), any())).willReturn(voucherEntitlement);
+        List<LocalDate> dateOfBirthsOfChildren = singletonList(LocalDate.now().minusMonths(8));
+        Optional<LocalDate> expectedDueDate = Optional.of(LocalDate.now().plusMonths(8));
+
+        PaymentCycleVoucherEntitlement result =
+                paymentCycleEntitlementCalculator.calculateEntitlement(expectedDueDate, dateOfBirthsOfChildren, LocalDate.now(), null);
```

Comment:
It would be worth refactoring this `null` into a variable called `PaymentCycleVoucherEntitlement previousVoucherEntitlement` as it took me a while to figure out where the previous entitlement was in this test, had to go to the coded to figure it out.

Reply:
Done